library/src/main/java/com/novoda/downloadmanager/lib/DownloadThread.java

```diff
@@ -485,15 +494,24 @@ private void checkConnectivity() throws StopRequestException {
             int status = DownloadStatus.WAITING_FOR_NETWORK;
             if (networkUsable == NetworkState.UNUSABLE_DUE_TO_SIZE) {
                 status = DownloadStatus.QUEUED_FOR_WIFI;
-                originalDownloadInfo.notifyPauseDueToSize(true);
+                notifyPauseDueToSize(true);
             } else if (networkUsable == NetworkState.RECOMMENDED_UNUSABLE_DUE_TO_SIZE) {
                 status = DownloadStatus.QUEUED_FOR_WIFI;
-                originalDownloadInfo.notifyPauseDueToSize(false);
+                notifyPauseDueToSize(false);
             }
             throw new StopRequestException(status, networkUsable.name());
         }
     }
 
+    void notifyPauseDueToSize(boolean isWifiRequired) {
+        Intent intent = new Intent(Intent.ACTION_VIEW);
+        intent.setData(originalDownloadInfo.getAllDownloadsUri());
+        intent.setClassName(SizeLimitActivity.class.getPackage().getName(), SizeLimitActivity.class.getName());
```

Comment:
maybe this could also sit on a listener on the client side, rather than the lib controlling the dialog?